go/interpreter/lfvm/ct/ct.go

```diff
@@ -64,47 +69,48 @@ func (a *ctAdapter) StepN(state *st.State, numSteps int) (*st.State, error) {
 	}
 
 	// Set up execution context.
-	var ctxt = &context{
-		pc:         int32(pcMap.evmToLfvm[state.Pc]),
-		params:     params,
-		context:    params.Context,
-		gas:        params.Gas,
-		refund:     tosca.Gas(state.GasRefund),
-		stack:      convertCtStackToLfvmStack(state.Stack),
-		memory:     memory,
-		status:     statusRunning,
-		code:       converted,
-		returnData: state.LastCallReturnData.ToBytes(),
-	}
+	var ctxt = lfvm.NewContext(
```

Comment:
IMO this is not better, 
We had named fields initialization and the new code has way too many arguments for a function call, including 3 new undocumented arguments at the end that could be defaulted or explained. 
If the goal is to prevent leaking private types, I would reconsider first if making the context public is that bad. 
If it cannot happen. Then I would remove any defaulted value from this signature like the three obscure last arguments, status, last call return data, etc... 

If this function is really unavoidable and still has more than 6 arguments, I would consider adding a single argument being a struct, so that we can have the named arguments back.